.scripts/check-alterations-sequence.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * This script runs a task to check alteration files seqence:
+ * new files should come last
+ *
+ */
+
+const { execSync } = require("child_process");
+
+const alterationFilePrefix = "packages/schemas/alterations/";
+
+const allAlterations = execSync("pnpm cli db alter list", {
+  encoding: "utf-8",
+})
+  .split("\n")
+  .filter((filename) => Boolean(filename))
+  .map((filename) => filename.replace(".js", ""));
+
+const diffFiles = execSync("git diff --name-only HEAD HEAD~1", {
+  encoding: "utf-8",
+});
+const committedAlterations = diffFiles
+  .split("\n")
+  .filter((filename) => filename.startsWith(alterationFilePrefix))
+  .map((filename) =>
+    filename.replace(alterationFilePrefix, "").replace(".ts", "")
+  );
+
+for (const alteration of committedAlterations) {
+  const index = allAlterations.indexOf(alteration);
+
+  if (index < allAlterations.length - committedAlterations.length) {
```

Comment:
Shouldn't we check the timestamp instead of committed index?

Reply:
The `allAlterations` is sorted, and this sorted method is the same as we run alteration deploy. The timestamp is calculated in internally.

Reply:
So when we get `allAlterations`, it is actually ordered by timestamp, then we only need to check the index.